lib/api.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * API Endpoints.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+/**
+ * Validates that a preference name is valid, as part of a REST API request.
+ *
+ * @param  string           $preference_name Preference name.
+ * @param  WP_REST_Request  $request The REST request.
+ * @param  string           $request The parameter key for the value being validated.
+ * @return bool             If the preference name is valid.
+ */
+function gutenberg_validate_preference_name( $preference_name, $request, $key ) {
```

Comment:
Do we need to wrap all of this logic in functions?  Why not just a single `static class` that also contains a `register_routes` method?